alvsan09/theia
# Asl/remove recently opened workspace

#### What it does


Fixes #4884
Remove from recently opened workspaces list
  
Added the ability to remove Recent Workspaces from the Open Recent
Workspace... quick-open command palette. At the moment, the only way
to be able to remove workspaces from the list was to completely
delete the recent workspace.json file found under the user's home
under .theia. This meant that users would not be able to trim their
list of outdated workspaces causing too much noise. Users can now
directly remove items from the list.

#### How to test

1) Open the command palette (Ctrl+Shft+P)
2) Type or select the command "File: Open recent workspace"
3) Using the mouse, hover over the different entries in the list and make sure to locate a `close` icon 'X' at the end of the item in focus.
4) Click the 'X' icon to remove the item from the list
5) Refresh the browser and make sure the workspace removed in the previous step does not appear

Other instances of quick access should not be impacted, try few e.g. from the main menu Terminal -> Run Task, and any other run command, etc.

![remove_from_recent_opened_workspaces](https://user-images.githubusercontent.com/76971376/126233757-a90f359d-1022-4bfa-918b-3525584aa183.gif)


#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
